### PR TITLE
chore!: default value to None for eval_frequency

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Related issue

- https://github.com/Substra/substra/issues/343

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: substrafl/evaluation_strategy.py [hunk 1]
```diff
@@ -18,10 +18,10 @@ def __init__(
 
         Args:
             test_data_nodes (List[TestDataNode]): nodes on which the model is to be tested.
-            eval_frequency (Optional[int]): The model will be tested every ``eval_frequency`` rounds. Default to 1.
-                Set to None to activate eval_rounds only.
+            eval_frequency (Optional[int]): The model will be tested every ``eval_frequency`` rounds.
+                Set to None to activate eval_rounds only. Default to None.
             eval_rounds (Optional[List[int]]): If specified, the model will be tested on the index of a round given
-                in the rounds list. Set to None to activate eval_frequency only.
+                in the rounds list. Set to None to activate eval_frequency only. Default to None.
```
Comment (tanguy-marchand) [inline substrafl/evaluation_strategy.py:24]: Shouldn't you raise an error if both values are left to None? And specify that in the doc?

File: substrafl/remote/register/register.py [hunk 2]
```diff
@@ -148,6 +148,7 @@ def _get_base_docker_image(python_major_minor: str, editable_mode: bool):
                 f"Your environment uses substra-tools={substratools_image_version}. Version \
                 {MINIMAL_DOCKER_SUBSTRATOOLS_VERSION} will be used on Docker.",
                 exceptions.SubstraToolsDeprecationWarning,
+                stacklevel=2,
```
Comment (tanguy-marchand) [inline substrafl/remote/register/register.py:151]: no idea what this part does

File: substrafl/evaluation_strategy.py (hunk 1 above)
Comment (ThibaultFy) [reply substrafl/evaluation_strategy.py:24]: It was already checked and present il the docstring. Cf `At least one of eval_frequency or eval_rounds must be defined.` and `ValueError: both eval_rounds and eval_frequency cannot be None at the same time`

File: substrafl/evaluation_strategy.py [hunk 3]
```diff
@@ -18,10 +18,10 @@ def __init__(
 
         Args:
             test_data_nodes (List[TestDataNode]): nodes on which the model is to be tested.
-            eval_frequency (Optional[int]): The model will be tested every ``eval_frequency`` rounds. Default to 1.
-                Set to None to activate eval_rounds only.
+            eval_frequency (Optional[int]): The model will be tested every ``eval_frequency`` rounds.
+                Set to None to activate eval_rounds only. Default to None.
```
Comment (oleobal) [inline substrafl/evaluation_strategy.py:22]: Nitpick: use the third person to address the variable, like on [line 121](https://github.com/Substra/substrafl/blob/d70bae2d105c6531a6fa0f71989edb7bf5bf52c0/substrafl/evaluation_strategy.py#L121).

```suggestion
                Set to None to activate eval_rounds only. Defaults to None.
```

This also goes for line 24.

**Pull request merged**